proxy/proxy_test.go

```diff
@@ -195,6 +213,17 @@ func TestProxyGRPC(t *testing.T) {
 			runGRPCTest(t, tc)
 		})
 	}
+
+	for i := 0; i < 20; i++ {
```

Comment:
What's the goal w/ this added iteration of the test case? Given that the iteration index (or no other new information) is actually propagated into the test context.